### PR TITLE
Introduce EditorConfig

### PR DESCRIPTION
**Pull request by szepeviktor**
- add EC configuration
- fix problems
- add CI steps
- fix warning about outdated actions/checkout

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+root = true
+
+[*]
+indent_style = space
+indent_size = 4
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+max_line_length = 200
```
Comment (szepeviktor) [inline .editorconfig:12]: I would set 120 for max line length but this is not my project!

Comment (szepeviktor) [reply .editorconfig:12]: 165 seem to be possible by shortening 3 lines!

**Comment by PrinsFrank** [comment]
Hey @szepeviktor, sorry I forgot about this PR! Can you updated it with the latest changes or should I do that?

**Comment by szepeviktor** [comment]
🤖 Resolving conflicts ...

**Review by PrinsFrank — APPROVED** [review]
Thanks!!

**Comment by szepeviktor** [comment]
Glad to meet your bytes.

**Pull request merged**